diff --git a/node_modules/vue-slider-component/lib/vue-slider-dot.vue b/node_modules/vue-slider-component/lib/vue-slider-dot.vue
index 455e822..cfbdece 100644
--- a/node_modules/vue-slider-component/lib/vue-slider-dot.vue
+++ b/node_modules/vue-slider-component/lib/vue-slider-dot.vue
@@ -2,7 +2,7 @@
   <div
     ref="dot"
     :class="dotClasses"
-    :aria-valuetext="tooltipValue"
+    :aria-valuetext="(tooltipValue as string)"
     @mousedown.passive="dragStart"
     @touchstart.passive="dragStart"
   >
